avro-serializer/src/main/java/io/confluent/kafka/serializers/KafkaAvroDecoder.java

```diff
@@ -28,12 +28,17 @@ public KafkaAvroDecoder(SchemaRegistryClient schemaRegistry) {
     this.schemaRegistry = schemaRegistry;
   }
 
+    public KafkaAvroDecoder(SchemaRegistryClient schemaRegistry, boolean useSpecificDecoder) {
```

Comment:
Ah, I see the issue. One alternative would be to still pass in the Properties object and refactor parsing of useSpecificDecoder to a separate method that can be reused.

I'm only being picky about this because this is technically a public interface even though it's just for testing and we need to be careful about compatibility. If we end up with a few more of these options, we might just end up with a bunch of different constructors to toggle them all + maintain backwards compatibility. It's definitely possible users will want to also take advantage of these constructors for their own unit testing, so getting the interface right is pretty important.